pallets/passkey/src/lib.rs

```diff
@@ -350,3 +362,63 @@ where
 		)
 	}
 }
+
+/// Frame system related checks
+#[derive(Encode, Decode, Clone, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct FrameSystemChecks<T: Config + Send + Sync>(pub T::AccountId, pub Call<T>);
+
+impl<T: Config + Send + Sync> FrameSystemChecks<T>
+where
+	<T as frame_system::Config>::RuntimeCall:
+		From<Call<T>> + Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
+{
+	/// Validates the transaction fee paid with tokens.
+	pub fn pre_dispatch(&self) -> Result<(), TransactionValidityError> {
+		let info = &self.1.get_dispatch_info();
+		let len = self.1.using_encoded(|c| c.len());
+		let runtime_call: <T as frame_system::Config>::RuntimeCall =
+			<T as frame_system::Config>::RuntimeCall::from(self.1.clone());
+		let who = self.0.clone();
+
+		let non_zero_sender_check = frame_system::CheckNonZeroSender::<T>::new();
+		let spec_version_check = frame_system::CheckSpecVersion::<T>::new();
+		let tx_version_check = frame_system::CheckTxVersion::<T>::new();
+		let genesis_hash_check = frame_system::CheckGenesis::<T>::new();
+		let era_check = frame_system::CheckEra::<T>::from(Era::immortal());
+
+		non_zero_sender_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		spec_version_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		tx_version_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		genesis_hash_check.pre_dispatch(&who, &runtime_call, info, len)?;
+		era_check.pre_dispatch(&who, &runtime_call, info, len)
```

Comment:
I just wanted to double check if we are missing a `?` operator here?

Reply:
No since this returns from this point 
> I just wanted to double check if we are missing a ? operator here?